### PR TITLE
Internal: Modules/Usage - Add page settings to tracking data [ED-1229]

### PR DESCRIPTION
**Pull request by iNewLegend**

### CODE REVIEW

**Review by matipojo — CHANGES_REQUESTED** [review]
Please remove the count of repeaters with default values.
For example, `system_colors` with 4 default colors should be counted as `0`

**Comment by iNewLegend** [comment]
Please check here https://github.com/elementor/elementor/pull/18622

**Review by matipojo — CHANGES_REQUESTED** [review]

File: core/base/document.php [hunk 1]
```diff
@@ -1425,6 +1426,101 @@ public function __construct( array $data = [] ) {
...
+			$element_controls = $element_instance->get_controls();
+
+			if ( isset( $element['settings'] ) ) {
+				$settings_controls = $element['settings'];
+				$element_ref = &$usage[ $type ];
+
+				// Add dynamic values.
+				$settings_controls = Usage::add_general_controls( $settings_controls, $element_ref );
+
+				$changed_controls_count = Usage::add_controls( $settings_controls, $element_controls, $element_ref );
```
Comment (matipojo) [inline core/base/document.php:1511]: Check why mobile padding didn't appear in the usage data

File: core/base/document.php [hunk 2]
```diff
@@ -1425,6 +1426,101 @@ public function __construct( array $data = [] ) {
 		parent::__construct( $data );
 	}
 
+	/**
+	 * Get document settings usage.
+	 *
+	 * @return array
+	 */
+	public function get_usage() {
```
Comment (matipojo) [inline core/base/document.php:1434]: On recalc clear all document usage data

File: core/base/document.php [hunk 3]
```diff
@@ -1425,6 +1426,101 @@ public function __construct( array $data = [] ) {
...
+		$usage  = [];
```
Comment (matipojo) [inline core/base/document.php:1435]: On recalc, check the notice: 
`Notice: Array to string conversion in /var/www/html/wp-content/plugins/elementor/includes/base/controls-stack.php on line 1465`

**Review by matipojo — CHANGES_REQUESTED** [review]
See comments

File: core/base/document.php (hunk 1 above)
Comment (iNewLegend) [reply core/base/document.php:1511]: image

File: core/base/document.php (hunk 2 above)
Comment (iNewLegend) [reply core/base/document.php:1434]: https://github.com/elementor/elementor/pull/18687/commits/a35c31f09cb41bafa16cfadff676e7c18c859b07

File: core/base/document.php (hunk 3 above)
Comment (iNewLegend) [reply core/base/document.php:1435]: https://github.com/elementor/elementor/pull/18687/commits/a35c31f09cb41bafa16cfadff676e7c18c859b07 - 
it caused by multidimensional array

File: includes/base/controls-stack.php [hunk 4]
```diff
@@ -1429,6 +1429,45 @@ public function is_control_visible( $control, $values = null ) {
...
+			 * So only 'color' will be analyzed.
+			 */
+			foreach ( $control_default as $key => $control_repeater_item_default ) {
+				$clear_repeater_settings[ $key ] = array_intersect_key(
+					$setting_value[ $key ],
+					array_flip( /* allowed keys*/ array_keys( $control_repeater_item_default ) )
+				);
+			}
+
+			$is_default = $clear_repeater_settings === $control_default || 0 === count( array_diff( $control_default, $clear_repeater_settings ) );
```
[the remark this answers is not in the document]
Comment (iNewLegend) [reply includes/base/controls-stack.php:1465]: https://github.com/elementor/elementor/pull/18687/commits/7d200c064e1e716221238d645e5da63e362be4e5
Im not sure about it.

File: core/base/document.php (hunk 1 above)
Comment (matipojo) [reply core/base/document.php:1511]: Please test with a heading that you changed the mobile padding

**Pull request closed without merging**